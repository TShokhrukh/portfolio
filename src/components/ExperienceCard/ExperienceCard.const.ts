import { cn } from '@bem-react/classname'

export const cnExperienceCard = cn('ExperienceCard')
