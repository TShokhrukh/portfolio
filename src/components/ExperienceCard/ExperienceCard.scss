.ExperienceCard {
  border: 1px solid rgba($purple, 0.5);
  border-radius: $border-radius * 2;
  padding: $spacer * 5.25 $spacer * 2.5 $spacer * 3.75 $spacer * 2.5;
  display: flex;
  flex-direction: column;
  align-items: center;
}