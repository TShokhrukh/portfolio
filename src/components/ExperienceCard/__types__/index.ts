export * from './ExperienceCard.types'
export * from './ExperienceCard-Icon.types'
