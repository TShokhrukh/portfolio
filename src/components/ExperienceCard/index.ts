export * from './build'
export * from './__types__'
export { ExperienceCard as ExperienceCardBase } from './ExperienceCard'
