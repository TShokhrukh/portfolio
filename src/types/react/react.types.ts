import type { ElementType } from 'react'

export interface IComponentProp<P = any> {
  component?: ElementType<P>,
}
