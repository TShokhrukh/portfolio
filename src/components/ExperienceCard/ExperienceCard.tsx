import React, { FC } from 'react'
import { ExperienceCardIcon } from './Icon'
import { ExperienceCardBody } from './Body'
import { ExperienceCardSkill } from './Skill'
import { cnExperienceCard } from './ExperienceCard.const'
import { Context } from './lib/context'

import type { IExperienceCardProps } from './__types__'

import './ExperienceCard.scss'

export const ExperienceCard: FC<IExperienceCardProps> = ({
  component: Component = 'div',
  className,
  experience,
  ...props
}) => {
  return (
    <Context.Provider value={experience}>
      <Component {...props} className={cnExperienceCard({}, [className])}>
        <ExperienceCardIcon />
        <ExperienceCardBody />
        <ExperienceCardSkill />
      </Component>
    </Context.Provider>
  )
}
