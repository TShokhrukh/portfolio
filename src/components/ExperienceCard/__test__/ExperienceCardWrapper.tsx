import React, { ReactNode } from 'react'
import { TranslationsProvider } from '@eo-locale/react'

import type { Locale } from '@eo-locale/core'

interface IWrapperProps {
  children: ReactNode
}

const locales: Locale[] = [
  {
    language: 'en',
    messages: {}
  }
]

export const ExperienceCardWrapper: React.FC<IWrapperProps> = ({ children }) => {
  return (
    <TranslationsProvider language={'en'} locales={locales} onError={() => null}>
      {children}
    </TranslationsProvider>
  )
}
