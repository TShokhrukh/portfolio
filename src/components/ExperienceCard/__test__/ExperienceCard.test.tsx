/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { ExperienceCard } from '../ExperienceCard'
import { ExperienceCardWrapper } from './ExperienceCardWrapper'

describe('ExperienceCard', () => {
  test('must return the component template', () => {
    const experience = {
      icon: 'home',
      locales: {
        en: {
          body: 'home_body',
          skill: 'home_skill'
        }
      }
    }

    render(
      <ExperienceCardWrapper>
        <ExperienceCard experience={experience} data-testid='experience_card' />
      </ExperienceCardWrapper>
    )

    expect(screen.getByTestId('experience_card')).toMatchSnapshot()
  })
})
