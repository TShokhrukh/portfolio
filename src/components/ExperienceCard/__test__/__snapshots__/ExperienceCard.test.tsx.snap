// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceCard must return the component template 1`] = `
<div
  class="ExperienceCard"
  data-testid="experience_card"
>
  <span
    aria-hidden="true"
    class="fa fa-home Icon ExperienceCard-Icon"
  />
  <p
    class="ExperienceCard-Body"
  >
    home_body
  </p>
  <h3
    class="ExperienceCard-Skill"
  >
    home_skill
  </h3>
</div>
`;
